### PR TITLE
Redux integration

### PR DESCRIPTION
**Pull request by pankajsoni91**
Bind the store directly to Dashboard. Do I need to create the different folder for all the container component?

Please review and I will give you the PR on a different branch for merge.

### CODE REVIEW

File: src/client/index.tsx [hunk 1]
```diff
@@ -2,7 +2,16 @@ import * as React from 'react';
 import { render } from 'react-dom';
 import App from './routes/Routes';
 import { BrowserRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import store from './redux/store/configureStore';
```
Comment (kuldeepkeshwar) [inline src/client/index.tsx]: store should be exported as singleton object

File: src/client/redux/reducers/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { combineReducers } from 'redux';
+import {
+  reducer as dashboardReducer,
+  IReducerState as IDashboardReducerState,
+} from '../../routes/dashboard/reducer';
```
Comment (kuldeepkeshwar) [inline src/client/redux/reducers/index.ts]: use relative path

File: src/client/redux/reducers/index.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export default combineReducers({
+  dashboard: dashboardReducer,
+});
+
+export interface IAppState {
+  test: IDashboardReducerState;
```
Comment (kuldeepkeshwar) [inline src/client/redux/reducers/index.ts]: `test`  should be `dashboard`

File: src/client/redux/store/configureStore.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { createStore, applyMiddleware, compose } from 'redux';
+import rootReducer from '../reducers';
+import { IAppState } from '../reducers';
```
Comment (kuldeepkeshwar) [inline src/client/redux/store/configureStore.ts]: use single import statement for importing from the same file

PS there is a tslint rule for this, please add tslint to project

File: src/client/routes/Routes.tsx [hunk 5]
```diff
@@ -1,18 +1,28 @@
 import * as React from 'react';
 import { Route, Link } from 'react-router-dom';
-import SubRoute from './test-route';
 import { hot } from 'react-hot-loader';
+import Home from './home';
```
Comment (kuldeepkeshwar) [inline src/client/routes/Routes.tsx]: component(file/folder) name should start with caps

File: src/client/routes/dashboard/index.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import { bindActionCreators } from 'redux';
+import { connect, Dispatch } from 'react-redux';
+import { IAppState } from '../../redux/reducers';
+import { actions } from './reducer';
+
+const Dashboard = () => {
+  return <div>Dashboard page</div>;
+};
+
+const mapStateToProps = (data: IAppState) => ({ data });
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx]: `data` should be state

File: src/client/index.tsx (hunk 1 above)
Comment (pankajsoni91) [reply src/client/index.tsx]: I think the name is misleading here. It should be configured store and I am calling the same which passing the value to store.

File: src/client/routes/dashboard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect, Dispatch } from 'react-redux';
+import { IAppState } from '../../redux/reducers';
+import { actions } from './reducer';
+
+const Dashboard = () => {
+  return <div>Dashboard page</div>;
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx]: please make use of props

File: src/client/redux/reducers/index.ts (hunk 3 above)
Comment (pankajsoni91) [reply src/client/redux/reducers/index.ts]: Done

File: src/client/routes/dashboard/index.tsx (hunk 6 above)
Comment (pankajsoni91) [reply src/client/routes/dashboard/index.tsx]: const mapStateToProps = ({dashboard}) => ({ dashboard });
As we are only need to bind the dashboard state here.

File: src/client/routes/dashboard/index.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+};
+
+const mapStateToProps = (data: IAppState) => ({ data });
+
+// TODO - need to check whether any need to be remove or not
+const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
+  actions: bindActionCreators(actions, dispatch)
```
Comment (kuldeepkeshwar) [inline src/client/routes/dashboard/index.tsx]: we should avoid pass actions like this, can cause readability issue

Comment (pankajsoni91) [reply src/client/routes/dashboard/index.tsx]: little help would be required over here.

**Pull request merged**